db/sg_replicate_cfg.go

```diff
@@ -248,7 +251,7 @@ func (rc *ReplicationConfig) Upsert(c *ReplicationUpsertConfig) {
 		rc.Username = *c.Username
 	}
 
-	if c.Username != nil {
+	if c.Username != nil && c.Password != nil {
```

Comment:
This change doesn't look right?

Upsert merges the given `c` into `rc` - which isn't going to happen if you try to just change a password without also re-stating the username.